client/src/app/core/ui-services/personal-note.service.ts

```diff
@@ -66,4 +66,38 @@ export class PersonalNoteService {
             await this.http.put(`rest/users/personal-note/${pnObject.id}/`, pnObject);
         }
     }
+
+    /**
+     * Sets the 'favorite' status for several models of a type in bulk
+     *
+     * @param models The model the content belongs to
+     * @param star The new 'favorite' status
+     */
+    public async bulkSetStar(models: BaseModel[] | BaseViewModel[], star: boolean): Promise<void> {
+        if (!models.length) {
+            throw new Error('No models selected');
+        }
+        const pnObject: Partial<PersonalNoteObject> = this.personalNoteObject || {};
+        if (!pnObject.notes) {
+            pnObject.notes = {};
+        }
+        const colString = models[0].collectionString;
+        if (!pnObject.notes[colString]) {
+            pnObject.notes[colString] = {};
+        }
+        for (const model of models) {
+            if (!star && pnObject.notes[colString][model.id] && pnObject.notes[colString][model.id].star) {
```

Comment:
Can be simplified to:
```
if (pnObject.notes[colString][model.id]) {
    pnObject.notes[colString][model.id].star = star;
} else {
    pnObject.notes[colString][model.id] = { star: star, note: '' };
}
```
So if there isn't a pnObject, create the default one. If there is one, we assume it is an object, so set the star attribute to the given value.